### PR TITLE
TUF `TrustUpdater` basic logging

### LINKED ISSUE #368: Improve (audit) logging of key/certificate material
TUF is now used to download the various certificates and public keys. It would be good to review the visibility of 
* which keys/certificates are used during a specific sigstore CLI command
* when certificates and keys are updated

Both of these could be beneficial for debugging and auditability.

I suppose a first step might be to ensure that we log the keys/certs we use at least at DEBUG level: the simplest thing might be for the TrustUpdater to log the hash (and maybe even the cached file location) of the artifacts that are returned via `TrustUpdater.get_*()`. It's possible that python-tuf already logs something like this but I am not sure.

### PR DESCRIPTION
**Pull request by jleightcap**
Stripping down prior branch to just one `logger.debug` call in `TrustUpdater`. Here, just log each of the cache items and their contained key.

Closes #368 

Example: `SIGSTORE_LOGLEVEL="debug" make run ARGS="verify ..."`

```sh
# ...
DEBUG:sigstore._internal.tuf:TUF metadata: /Users/jleightcap/Library/Application Support/sigstore-python/tuf/https%3A%2F%2Fsigstore-tuf-root.storage.googleapis.com%2F
DEBUG:sigstore._internal.tuf:TUF targets cache: /Users/jleightcap/Library/Caches/sigstore-python/tuf/https%3A%2F%2Fsigstore-tuf-root.storage.googleapis.com%2F
DEBUG:sigstore._internal.tuf:TUF cache target Rekor ['Active']: rekor.pub with hash
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
-----END PUBLIC KEY-----

DEBUG:sigstore._internal.tuf:TUF cache target CTFE ['Active']: ctfe.pub with hash
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEbfwR+RJudXscgRBRpKX1XFDy3Pyu
dDxz/SfnRi1fT8ekpfBd2O1uoz7jr3Z8nKzxA69EUQ+eFCFI3zeubPWU7w==
-----END PUBLIC KEY-----
DEBUG:sigstore._internal.tuf:TUF cache target CTFE ['Active']: ctfe_2022.pub with hash
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEiPSlFi0CmFTfEjCUqF9HuCEcYXNK
AaYalIJmBZ8yyezPjTqhxrKBpMnaocVtLJBI1eM3uXnQzQGAJdJ4gs9Fyw==
-----END PUBLIC KEY-----

DEBUG:sigstore._internal.tuf:TUF cache target Fulcio ['Active', 'Expired']: fulcio.crt.pem with hash
-----BEGIN CERTIFICATE-----
MIIB+DCCAX6gAwIBAgITNVkDZoCiofPDsy7dfm6geLbuhzAKBggqhkjOPQQDAzAq
MRUwEwYDVQQKEwxzaWdzdG9yZS5kZXYxETAPBgNVBAMTCHNpZ3N0b3JlMB4XDTIx
MDMwNzAzMjAyOVoXDTMxMDIyMzAzMjAyOVowKjEVMBMGA1UEChMMc2lnc3RvcmUu
ZGV2MREwDwYDVQQDEwhzaWdzdG9yZTB2MBAGByqGSM49AgEGBSuBBAAiA2IABLSy
A7Ii5k+pNO8ZEWY0ylemWDowOkNa3kL+GZE5Z5GWehL9/A9bRNA3RbrsZ5i0Jcas
taRL7Sp5fp/jD5dxqc/UdTVnlvS16an+2Yfswe/QuLolRUCrcOE2+2iA5+tzd6Nm
MGQwDgYDVR0PAQH/BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQEwHQYDVR0OBBYE
FMjFHQBBmiQpMlEk6w2uSu1KBtPsMB8GA1UdIwQYMBaAFMjFHQBBmiQpMlEk6w2u
Su1KBtPsMAoGCCqGSM49BAMDA2gAMGUCMH8liWJfMui6vXXBhjDgY4MwslmN/TJx
Ve/83WrFomwmNf056y1X48F9c4m3a3ozXAIxAKjRay5/aj/jsKKGIkmQatjI8uup
Hr/+CxFvaJWmpYqNkLDGRU+9orzh5hI2RrcuaQ==
-----END CERTIFICATE-----
DEBUG:sigstore._internal.tuf:TUF cache target Fulcio ['Active', 'Expired']: fulcio_intermediate_v1.crt.pem with hash
-----BEGIN CERTIFICATE-----
MIICGjCCAaGgAwIBAgIUALnViVfnU0brJasmRkHrn/UnfaQwCgYIKoZIzj0EAwMw
KjEVMBMGA1UEChMMc2lnc3RvcmUuZGV2MREwDwYDVQQDEwhzaWdzdG9yZTAeFw0y
MjA0MTMyMDA2MTVaFw0zMTEwMDUxMzU2NThaMDcxFTATBgNVBAoTDHNpZ3N0b3Jl
LmRldjEeMBwGA1UEAxMVc2lnc3RvcmUtaW50ZXJtZWRpYXRlMHYwEAYHKoZIzj0C
AQYFK4EEACIDYgAE8RVS/ysH+NOvuDZyPIZtilgUF9NlarYpAd9HP1vBBH1U5CV7
7LSS7s0ZiH4nE7Hv7ptS6LvvR/STk798LVgMzLlJ4HeIfF3tHSaexLcYpSASr1kS
0N/RgBJz/9jWCiXno3sweTAOBgNVHQ8BAf8EBAMCAQYwEwYDVR0lBAwwCgYIKwYB
BQUHAwMwEgYDVR0TAQH/BAgwBgEB/wIBADAdBgNVHQ4EFgQU39Ppz1YkEZb5qNjp
KFWixi4YZD8wHwYDVR0jBBgwFoAUWMAeX5FFpWapesyQoZMi0CrFxfowCgYIKoZI
zj0EAwMDZwAwZAIwPCsQK4DYiZYDPIaDi5HFKnfxXx6ASSVmERfsynYBiX2X6SJR
nZU84/9DZdnFvvxmAjBOt6QpBlc4J/0DxvkTCqpclvziL6BCCPnjdlIB3Pu3BxsP
mygUY7Ii2zbdCdliiow=
-----END CERTIFICATE-----
DEBUG:sigstore._internal.tuf:TUF cache target Fulcio ['Active', 'Expired']: fulcio_v1.crt.pem with hash
-----BEGIN CERTIFICATE-----
MIIB9zCCAXygAwIBAgIUALZNAPFdxHPwjeDloDwyYChAO/4wCgYIKoZIzj0EAwMw
KjEVMBMGA1UEChMMc2lnc3RvcmUuZGV2MREwDwYDVQQDEwhzaWdzdG9yZTAeFw0y
MTEwMDcxMzU2NTlaFw0zMTEwMDUxMzU2NThaMCoxFTATBgNVBAoTDHNpZ3N0b3Jl
LmRldjERMA8GA1UEAxMIc2lnc3RvcmUwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAT7
XeFT4rb3PQGwS4IajtLk3/OlnpgangaBclYpsYBr5i+4ynB07ceb3LP0OIOZdxex
X69c5iVuyJRQ+Hz05yi+UF3uBWAlHpiS5sh0+H2GHE7SXrk1EC5m1Tr19L9gg92j
YzBhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRY
wB5fkUWlZql6zJChkyLQKsXF+jAfBgNVHSMEGDAWgBRYwB5fkUWlZql6zJChkyLQ
KsXF+jAKBggqhkjOPQQDAwNpADBmAjEAj1nHeXZp+13NWBNa+EDsDP8G1WWg1tCM
WP/WHPqpaVo0jhsweNFZgSs0eE7wYI4qAjEA2WB9ot98sIkoF3vZYdd3/VtWB5b9
TNMea7Ix/stJ5TfcLLeABLE4BNJOsQ4vnBHJ
-----END CERTIFICATE-----
# ...
```

### CODE REVIEW

File: sigstore/_internal/tuf.py [hunk 1]
```diff
@@ -152,7 +153,12 @@ def _get(self, usage: str, statuses: list[str]) -> list[bytes]:
                 if path is None:
                     path = self._updater.download_target(target_info)
                 with open(path, "rb") as f:
-                    data.append(f.read())
+                    target_hash = f.read()
```
Comment (woodruffw) [inline sigstore/_internal/tuf.py:156]: Where does the `target_hash` variable name come from? If I'm reading this right this variable actually holds the TUF targets' raw contents, not a hash.

File: sigstore/_internal/tuf.py [hunk 2]
```diff
@@ -156,7 +157,12 @@ def _get(self, usage: str, statuses: list[str]) -> list[bytes]:
                 if path is None:
                     path = self._updater.download_target(target_info)
                 with open(path, "rb") as f:
-                    data.append(f.read())
+                    target_cert = f.read()
+                    logger.debug(
+                        f"TUF cache target {usage} {statuses}: {os.path.basename(path)} with hash\n"
+                        f"{target_cert.decode('utf-8')}"
+                    )
+                    data.append(target_cert)
```
Comment (woodruffw) [inline sigstore/_internal/tuf.py:165]: These can really be anything (certs, bare public keys, etc.), so I'm going to rename it to `target_contents`.

```suggestion
                    target_contents = f.read()
                    logger.debug(
                        f"TUF cache target {usage} {statuses}: {os.path.basename(path)} with hash\n"
                        f"{target_contents.decode('utf-8')}"
                    )
                    data.append(target_contents)
```

File: sigstore/_internal/tuf.py (hunk 1 above)
Comment (jku) [reply sigstore/_internal/tuf.py]: If you refer to my idea in the original issue:

> the simplest thing might be for the TrustUpdater to log the hash (and maybe even the cached file location) of the artifacts

If that seems like a good idea, then you can just access `target_info`: it is of type [TargetFile](https://theupdateframework.readthedocs.io/en/latest/api/tuf.api.metadata.supporting.html#tuf.api.metadata.TargetFile) -- maybe something like `f"Updated {target_info.path}: {target_info.hashes}"` would work as a log line?

**Review by woodruffw — APPROVED** [review]

File: sigstore/_internal/tuf.py (hunk 1 above)
Comment (jku) [reply sigstore/_internal/tuf.py]: LGTM -- although I don't remember if all targets are text files... if they are not, then the decode may fail

Comment (woodruffw) [reply sigstore/_internal/tuf.py]: I believe they all are, although maybe that's a bad future-looking assumption for us to make. Maybe we should limit debug logging to targets that should always be text?

Comment (woodruffw) [reply sigstore/_internal/tuf.py]: Yeah, all TUF targets are (conceptually) "files": a target is just a name and an associated bag of bytes.

The real underlying issue here (I think) is that not all TUF targets are guaranteed to be text; some might be binary instead. But that shouldn't be an issue with anything we currently attempt to retrieve for Sigstore purposes.

**Pull request merged**